prisma/schema.prisma

```diff
@@ -17,10 +17,11 @@ datasource db {
 
 model User {
   id       Int         @id @default(autoincrement())
-  name     String
+  name     String?
   email    String      @unique
-  username String      @unique
-  roles    Roles[]
+  username String?     @unique
+  avatar   String?
```

Comment:
Same here. If we dont find a avatar we should use a default image.

Usually optional in a DB is bad

Reply:
I'm not really sure on how to do it.

It's a string, I thought I could add a default user image to the `public/images` dir and use the relative path as default src, but that didn't work.

What if we use an empty string as default image and check whether the value stored in the DB is falsey or not? An empty string returns false when we use the `??` operator, for example.

Reply:
The problem with this is that you will need 2 steps:
1. get db data
2. check db data

This will happen every time we want to show the user profile. If we can ensure the DB will always have one image, we only need to do this check once, when the user is created.

Besides that, searching data in a sparse database requires more work, making our queries slower.

Check these links:
- https://sisudata.com/glossary/what-is-sparse-data#:~:text=Sparse%20data%20is%20a%20variable,any%20of%20the%20values%20are.
- https://www.educba.com/sparse-matrix-in-python/

Reply:
Your idea of having a default image looks good, but I don't know if it works with relative paths. 
You'll probably need to have the entire URL to store it (but I not sure, we need to check it first)